src/core/lib/event_engine/posix_engine/ev_poll_posix.cc

```diff
@@ -811,11 +810,18 @@ Poller::WorkResult PollPoller::Work(EventEngine::Duration timeout) {
   mu_.Unlock();
   if (pending_events.empty()) {
     if (was_kicked_ext) {
-      return Poller::Kicked{};
+      return Poller::WorkResult::kKicked;
     }
-    return Poller::DeadlineExceeded{};
+    return Poller::WorkResult::kDeadlineExceeded;
+  }
+  // Invoke the provided callback.
+  poll_again();
```

Comment:
Given that a scheduler (async task executor) is already available here, why not make `poll_again` synchronous and put it on the scheduler? See the IOCP implementation, which would already prefer that. This simplifies the callback a bit, and groups the concurrency logic a bit better (visually, at least).

If you go that route, poll_again would either need to be an AnyInvocable or Closure, or the docstring would need to make it known that the function being referenced must outlive the call to `Work`. I'd prefer the former, which lets the type system enforce the constraint.

Reply:
Done.